l5kit/l5kit/data/proto_api.py

```diff
@@ -0,0 +1,178 @@
+
+
+class ProtoAPI:
+    def __init__(self, protobuf_map_path: str, pose_to_ecef: np.ndarray):
+        """
+        Interface to the raw protobuf map file with the following features:
+        - access to element using ID is O(1);
+        - access to coordinates in world ref system for a set of elements is O(1) after first access (lru cache)
+        - object support iteration using __getitem__ protocol
+
+        Args:
+            protobuf_map_path (str): path to the protobuf file
+            pose_to_ecef (np.ndarray): transformation matrix from world coordinates to ECEF (dataset dependent)
+        """
+        self.protobuf_map_path = protobuf_map_path
+        self.ecef_to_pose = np.linalg.inv(pose_to_ecef)
+
+        with open(protobuf_map_path, "rb") as infile:
+            mf = MapFragment()
+            mf.ParseFromString(infile.read())
+
+        self.elements = mf.elements
+        self.ids_to_el = {self.get_element_id(el): idx for idx, el in enumerate(self.elements)}  # fast lookup table
+
+    @staticmethod
+    @no_type_check
+    def get_element_id(element: MapElement) -> str:
+        """
+        Get the element id as a string.
+        Elements ids are stored as a variable len sequence of bytes in the protobuf
+
+        Args:
+            element (MapElement): the element in the protobuf
+
+        Returns:
+            str: the id as a str
+        """
+        return element.id.id.decode("utf-8")
+
+    @no_type_check
+    def unpack_deltas_cm(self, dx: Sequence[int], dy: Sequence[int], dz: Sequence[int], frame: GeoFrame) -> np.ndarray:
+        """
+        Get coords in world reference system (local ENU->ECEF->world).
+        See the protobuf annotations for additional information about how coordinates are stored
+
+        Args:
+            dx (Sequence[int]): X displacement in centimeters in local ENU
+            dy (Sequence[int]): Y displacement in centimeters in local ENU
+            dz (Sequence[int]): Z displacement in centimeters in local ENU
+            frame (GeoFrame): geo-location information for the local ENU. It contains lat and long origin of the frame
+
+        Returns:
+            np.ndarray: array of shape (Nx3) with XYZ coordinates in world ref system
+
+        """
+        x = np.cumsum(np.asarray(dx) / 100)
+        y = np.cumsum(np.asarray(dy) / 100)
+        z = np.cumsum(np.asarray(dz) / 100)
+        xyz = np.stack(pm.enu2ecef(x, y, z, frame.origin.lat_e7 * 1e-7, frame.origin.lng_e7 * 1e-7, 0), axis=-1)
+        xyz = transform_points(xyz, self.ecef_to_pose)
+        return xyz
+
+    @staticmethod
+    @no_type_check
+    def is_lane(element: MapElement) -> bool:
+        """
+        Check whether an element is a valid lane
+
+        Args:
+            element (MapElement): a proto element
+
+        Returns:
+            bool: True if the element is a valid lane
+        """
+        return bool(element.element.HasField("lane"))
+
+    @lru_cache(maxsize=CACHE_SIZE)
```

Comment:
`LANE_COORDS_CACHE_SIZE`

Reply:
this value is also used by crosswalk, do you want me to add a constant for each one? otherwise we can live it like this I think